src/Workspaces/Core/Portable/FindSymbols/LinkedFileReferenceLocationEqualityComparer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols;
+
+/// <summary>
+/// Helper comparer to enable consumers of <see cref="SymbolFinder.FindReferencesAsync(ISymbol, Solution,
+/// CancellationToken)"/> to process references found in linked files only a single time.
+/// </summary>
+internal sealed class LinkedFileReferenceLocationEqualityComparer : IEqualityComparer<ReferenceLocation>
+{
+    public static readonly IEqualityComparer<ReferenceLocation> Instance = new LinkedFileReferenceLocationEqualityComparer();
+
+    private LinkedFileReferenceLocationEqualityComparer()
+    {
+    }
+
+    public bool Equals([AllowNull] ReferenceLocation x, [AllowNull] ReferenceLocation y)
```

Comment:
>AllowNull [](http://example.com/codeflow?start=24&length=9)

Learning opportunity here, how is this better than just using "ReferenceLocation?" ?

Reply:
This is a struct, how can it be null?

Reply:
this was just how implement-interface spit htings out.  clearly that needs to be fixed.  will manually fix thsi up though :)

Reply:
fixed.